numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -4702,7 +4702,7 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
     DUAL_INHERIT(CDouble, Complex, ComplexFloating);
     SINGLE_INHERIT(CLongDouble, ComplexFloating);
 
-    DUAL_INHERIT2(String, String, Character);
+    DUAL_INHERIT2(String, Bytes, Character);
```

Comment:
Subtle: this creates a base type inheritance on `PyBytesType`. The replacement was done previously via `#define PyString_Type PyBytes_Type`